api/desecapi/tests/base.py

```diff
@@ -779,13 +779,22 @@ def requests_desec_domain_creation(cls, name=None):
         ]
 
     @classmethod
-    def requests_desec_domain_deletion(cls, name=None):
-        return [
-            cls.request_pdns_zone_delete(name=name, ns='LORD'),
-            cls.request_pdns_zone_delete(name=name, ns='MASTER'),
+    def requests_desec_domain_deletion(cls, domain):
+        requests = [
+            cls.request_pdns_zone_delete(name=domain.name, ns='LORD'),
+            cls.request_pdns_zone_delete(name=domain.name, ns='MASTER'),
             cls.request_pdns_update_catalog(),
         ]
 
+        if domain.is_locally_registrable:
+            delegate_at = cls._find_auto_delegation_zone(domain.name)
+            requests += [
+                cls.request_pdns_zone_update(name=delegate_at),
+                cls.request_pdns_zone_axfr(name=delegate_at),
+            ]
+
+        return requests
+
     @classmethod
     def requests_desec_domain_creation_auto_delegation(cls, name=None):
```

Comment:
same for creation? 😬

Reply:
no, because during creation time, we don't have an object (only the name)